packages/dashboard/src/DashboardLayout.tsx

```diff
@@ -0,0 +1,254 @@
+import React, {
+  ComponentType,
+  ReactElement,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import GoldenLayout, { ItemConfigType } from '@deephaven/golden-layout';
+import Log from '@deephaven/log';
+import { usePrevious } from '@deephaven/react-hooks';
+import { Provider, useDispatch, useSelector, useStore } from 'react-redux';
+import PanelManager, { ClosedPanels } from './PanelManager';
+import PanelErrorBoundary from './PanelErrorBoundary';
+import LayoutUtils from './layout/LayoutUtils';
+import {
+  dehydrate as dehydrateDefault,
+  hydrate as hydrateDefault,
+} from './DashboardUtils';
+import PanelEvent from './PanelEvent';
+import { GLPropTypes, useListener } from './layout';
+import { getDashboardData, updateDashboardData } from './redux';
+
+export type DashboardLayoutConfig = ItemConfigType[];
+
+const log = Log.module('DashboardLayout');
+
+const EMPTY_OBJECT = Object.freeze({});
+
+const DEFAULT_LAYOUT_CONFIG: DashboardLayoutConfig = [];
+
+const DEFAULT_CALLBACK = () => undefined;
+
+type DashboardData = {
```

Comment:
It seems this should be somehow dependent on the registered plugins. I.e. contain `links` for the `Linker` plugin, etc.